src/lib/ffi/ffi_cipher.cpp

```diff
@@ -36,37 +36,6 @@ struct botan_cipher_struct final : public botan_struct<Botan::Cipher_Mode, 0xB4A
       size_t m_ideal_update_size;
 };
 
-namespace {
-
-size_t ffi_choose_update_size(Botan::Cipher_Mode& mode) {
-   const size_t update_granularity = mode.update_granularity();
-   const size_t minimum_final_size = mode.minimum_final_size();
-
-   /*
-   * Return the minimum possible granularity given the FFI API constraints that
-   * we require the returned size be > minimum final size.
```

Comment:
Does this condition no longer hold? I would think this would break callers.

Reply:
To the best of my understanding it doesn't hold anymore (after the introduction of #3951). For instance: an AEAD that allows single-byte updates, could now be used via `botan_cipher_update()` with single-byte buffers. When trying to finalize the cipher with a single-byte bulffer, it will return a `BOTAN_FFI_ERROR_INSUFFICIENT_BUFFER_SPACE` and report the size of the required buffer via the out-variable. Repeating the finalize-call with a big-enough buffer will work.

Nevertheless, I totally agree that this could potentially break callers and force them to add similar handling as I explained above. See also the discussion here: https://github.com/randombit/botan/issues/4090#issuecomment-2144841857.

Reply:
FYI: #4122 is an alternative that keeps this condition alive. ... to the best of my understanding.